### PR TITLE
revert the change the error_code field in helics3 to prevent major is…

### PR DESCRIPTION
**Pull request by phlptp**
…sues for HELICS2 to 3 conversions

### Summary

If merged this pull request will revert the change to the error object used in the C shared API to prevent issues with the interfaces trying to provide a seamless transition from 2 to 3

### CODE REVIEW

File: src/helics/shared_api_library/backup/helics_api.h [hunk 1]
```diff
@@ -9,151 +9,154 @@ SPDX-License-Identifier: BSD-3-Clause
...
-    HELICS_CORE_TYPE_TCP = 6,
-    HELICS_CORE_TYPE_UDP = 7,
-    HELICS_CORE_TYPE_ZMQ_SS = 10,
-    HELICS_CORE_TYPE_NNG = 9,
-    HELICS_CORE_TYPE_TCP_SS = 11,
-    HELICS_CORE_TYPE_HTTP = 12,
-    HELICS_CORE_TYPE_WEBSOCKET = 14,
-    HELICS_CORE_TYPE_INPROC = 18,
-    HELICS_CORE_TYPE_NULL = 66
+   HELICS_CORE_TYPE_DEFAULT = 0,
```
Comment (kdheepak) [inline src/helics/shared_api_library/backup/helics_api.h:12]: Why is this 4 spaces changed to 3 spaces?

File: src/helics/shared_api_library/backup/helics_api.h [hunk 2]
```diff
@@ -193,41 +196,41 @@ const HelicsBool HELICS_TRUE = 1;
...
-    double real;
-    double imag;
+   double real;
+   double imag;
 } HelicsComplex;
 
 typedef struct HelicsError {
-    int32_t errorCode;
-    const char* message;
+   int32_t errorCode;
```
Comment (kdheepak) [inline src/helics/shared_api_library/backup/helics_api.h:232]: This `errorCode` still hasn't been changed?

Comment (phlptp) [reply src/helics/shared_api_library/backup/helics_api.h]: updated the file without regenerating it first,  should be updated now

File: src/helics/shared_api_library/backup/helics_api.h (hunk 1 above)
Comment (phlptp) [reply src/helics/shared_api_library/backup/helics_api.h]: this was fixed by formatting

**Comment by phlptp** [comment]
@kdheepak  I think I resolved the issues you raised

**Comment by phlptp** [comment]
@kdheepak  can you take another look at this

**Comment by kdheepak** [comment]
Does this need to be changed too?

https://github.com/GMLC-TDC/HELICS/blob/29d68284a47e64f4e47a4e7d6af0d1cbbe005f81/interfaces/octave/octave_maps.i#L9

**Comment by kdheepak** [comment]
```
interfaces/octave/octave_maps.i
9:switch (err->errorCode)
55:if ($1->errorCode!=HELICS_OK)

interfaces/java/java_maps.i
79:if ($1->errorCode!=HELICS_OK)
```

**Review by kdheepak — APPROVED** [review]

**Comment by phlptp** [comment]
@kdheepak  good catch,  made the changes

**Pull request merged**